torch/csrc/jit/mobile/model_compatibility.cpp

```diff
@@ -179,6 +187,69 @@ std::unordered_map<std::string, OperatorInfo> _get_model_ops_and_info(
   return result;
 }
 
+/********************** Get Type Table **********************/
+
+// Forward declare
+std::unordered_set<std::string> _get_model_contained_types(
+    const std::vector<IValue>& bytecode_ivalues);
+
+std::unordered_set<std::string> _get_model_contained_types(std::istream& in) {
+  std::unique_ptr<IStreamAdapter> rai = std::make_unique<IStreamAdapter>(&in);
+  return _get_model_contained_types(std::move(rai));
+}
+
+std::unordered_set<std::string> _get_model_contained_types(
+    const std::string& filename) {
+  std::unique_ptr<FileAdapter> rai = std::make_unique<FileAdapter>(filename);
+  return _get_model_contained_types(std::move(rai));
+}
+
+std::unordered_set<std::string> _get_model_contained_types(
+    std::shared_ptr<ReadAdapterInterface> rai) {
+  if (!check_zip_file(rai)) {
+    TORCH_CHECK(
+        false,
+        "Failed to open .ptl file please ensure the model was exported for mobile");
+  }
+  PyTorchStreamReader reader(std::move(rai));
+  auto bytecode_values = get_bytecode_ivalues(reader);
+  return _get_model_contained_types(bytecode_values);
+}
+
+// Get deduplicate type table given bytecode, and each string is a atomic type,
+// like str, Tensor and etc.
+std::unordered_set<std::string> _get_model_contained_types(
```

Comment:
Could you add an example in the comment, to show the bytecode, and the output of this function?